docs/rfcs/rfc-018-mathematical-operators.md

```diff
@@ -0,0 +1,167 @@
+# Mathematical operators
+
+## Problem
+As described in the issue #677, if we have to do a mathematical operation using the GraphQL library, 
+the common approach is to query the database and then perform the mutation with the final property value.
+The former approach has several downsides:
+- It requires multiple roundtrips to the database.
+- It's not concurrency safe.
+- It requires more code than necessary in the client.
+
+## Proposed Solution
+Add new input properties that solve atomic operations like:
+- `_INCREMENT`
+- `_DECREMENT`
+
+available for Int and BigInt fields.
+And:
+- `_ADD`
+- `_SUBTRACT`
+- `_MULTIPLY`
+- `_DIVIDE`
+
+available for Float fields.
+
+For instance `counter_ADD` or `counter_SUBTRACT`.
+These properties should be available for all the following types:
+- Int
+- BigInt
+- Float
+
+The MULTIPLY and the DIVIDE operations will then be available only for fields defined as Float.
+
+### Usage Examples
+Given the following type definitions:
+```graphql
+type Node {
+  id: ID! @id
+  counter: Int
+}
+```
+The operator _ADD could then be used in a mutation like:
+```graphql
+updateNodes(
+  where: { id: "e9bc687a-efd1-419d-b208" }
+  update: {
+    counter_INCREMENT: 1,
+  }
+)
+```
+
+### Technical considerations
+#### Ambiguous property
+In the following mutation body: 
+```graphql
+updateNodes(
+  where: { id: "e9bc687a-efd1-419d-b208" }
+  update: {
+    counter: 10,
+    counter_INCREMENT: 1,
+  }
+)
+```
+`counter` is ambiguous, and an Error will be raised.
+
+#### Schema size
+The proposed solution could increase the size of the augmented schema noticeably.
+
+#### Overflow
+In cases where a mathematical operation leads to an overflow, then an exception must be raised.
+
+## Alternative Solution
```

Comment:
We should move this to discarded solutions

Reply:
Done